### DISCUSSION TITLE
Connection to network devices over jump host

### DISCUSSION THREAD

**Discussion by avjay96** [ktbyers/netmiko#2654] [Q&A]
Hi, 
I need to establish connection to network devices which are behind the bastion host. I can access the bastion host using username and a password. I have tried the solution provided in the article [https://pynet.twb-tech.com/blog/automation/netmiko-proxy.html] but I couldn't establish connection to the network device. Is there a way to configure the ssh config file by providing the jump host username and password? Any help would really be appreciated.

**Comment by ktbyers** [comment]
@avjay96 Is there some reason you can't setup and use SSH key-based authentication for that first part of the SSH connection (i.e. for the connection between your machine and the bastion server)?

With Netmiko you could first connect from your local automation machine to the intermediate host (using username/password) and then from their issue an SSH command to connect to the end device. If you search on Netmiko and redispatch you should see some examples of this.

If you want to do this using an SSH-config file, then you would have to research whether you can use username/password for that first leg. It would be less secure, but you would have to research if it is allowed by ssh_config and how to do it.

**Reply by brandomando** [reply]
I was able to solve for this by opening the first connection to the bastion using the credentials provided to the factory function `ConnectHandler`, and then prior to opening the second connection and redispatching, changing the `username` and `password` attributes on the connection. Using Nornir, I just swapped them in and out of the `data` attribute of the inventory host for convenience.
